### PR TITLE
Refactoring of probe main

### LINKED ISSUE #537: Break up probe.main into smaller, testable functions.

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes #537

- Move probe/main.go to prog/probe/
- Rename probe/* the 'probe' package
- Moves resolver.go to xfer/
- Removed probe/instrumentation.go, not used anywhere
- An empty StingSet (and IDList) is now nil
- Move filter tests to filters_test.go

### CODE REVIEW

File: xfer/resolver.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package main
+package xfer
```
Comment (paulbellamy) [inline xfer/resolver.go]: Why the move of this?

**Comment by paulbellamy** [comment]
Let's break out the functional changes (StringSet stuff) from the re-org, just a separate commit would be nice.

File: xfer/resolver.go (hunk 1 above)
Comment (tomwilkie) [inline xfer/resolver.go]: It just didn't seem to 'fit' in the probe package; it was much more xfer-y, don't you think?

In other words: no good reason.

**Comment by tomwilkie** [comment]
+1 will do.

**Comment by tomwilkie** [comment]
@paulbellamy I think I've addressed your comment(s)

File: xfer/resolver.go (hunk 1 above)
Comment (paulbellamy) [inline xfer/resolver.go]: It's only ever used by the probe, though. So I feel it is much more probe-y...

Comment (tomwilkie) [inline xfer/resolver.go]: The same could be said for {Multi,Background,HTTP}Publishers and (Multi)AppClient...  The resolver should very much be colocated this there 5 things; I'm open to arguments that these should be in a probe/ sub module...

Comment (paulbellamy) [inline xfer/resolver.go]: after f2f discussion, lets move all this into the probe.

Comment (tomwilkie) [inline xfer/resolver.go]: We'll follow up with a ticket...

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**